framework/Models/HybridModel.py

```diff
@@ -0,0 +1,621 @@
+      unknownList = utils.checkIfUnknowElementsinList(modelOutputs, romOutputs)
+      if unknownList:
+        self.raiseAnError(IOError, 'Output Parameters: "', ','.join(str(e) for e in unknownList), '" used in ROM ', romIn.name, ' can not found in Model ', modelName)
+      if romIn.amITrained:
+        # Only untrained roms are allowed
+        self.raiseAnError(IOError,'HybridModel only accepts untrained ROM, but rom "', romIn.name, '" is already trained')
+
+    # check: we require that the union of ROMs outputs is the same as the paired model in order to use the ROM
+    # to replace the paired model.
+    if len(set(totalRomOutputs)) != len(totalRomOutputs):
+      dup = []
+      for elem in set(totalRomOutputs):
+        if totalRomOutputs.count(elem) > 1:
+          dup.append(elem)
+      # we assume there is no duplicate outputs among the roms
+      self.raiseAnError(IOError, 'The following outputs ', ','.join(str(e) for e in dup), "are found in the outputs of multiple roms!")
+    unknownList = utils.checkIfUnknowElementsinList(totalRomOutputs,modelOutputs)
+    if unknownList:
+      self.raiseAnError(IOError, "The following outputs: ", ','.join(str(e) for e in unknownList), " used in Model: ", modelName, "but not used in the paired ROMs.")
+
+  def getInitParams(self):
+    """
+      Method used to export to the printer in the base class the additional PERMANENT your local class have
+      @ In, None
+      @ Out, tempDict, dict, dictionary to be updated. {'attribute name':value}
+    """
+    tempDict = OrderedDict()
+    tempDict['ROMs contained in HybridModel are '] = self.romsDictionary.keys()
+    return tempDict
+
+  def getAdditionalInputEdits(self,inputInfo):
+    """
+      Collects additional edits for the sampler to use when creating a new input. In this case, it calls all the getAdditionalInputEdits methods
+      of the sub-models
+      @ In, inputInfo, dict, dictionary in which to add edits
+      @ Out, None.
+    """
+    self.modelInstance.getAdditionalInputEdits(inputInfo)
+
+  def __selectInputSubset(self,romName, kwargs):
+    """
+      Method aimed to select the input subset for a certain model
+      @ In, romName, string, the rom name
+      @ In, kwargs , dict, the kwarded dictionary where the sampled vars are stored
+      @ Out, selectedKwargs , dict, the subset of variables (in a swallow copy of the kwargs dict)
+    """
+    selectedKwargs = copy.copy(kwargs)
+    selectedKwargs['SampledVars'], selectedKwargs['SampledVarsPb'] = {}, {}
+    for key in kwargs["SampledVars"].keys():
+      if key in self.romsDictionary[romName]['Instance'].getInitParams()['Features']:
+        selectedKwargs['SampledVars'][key] = kwargs["SampledVars"][key]
+        selectedKwargs['SampledVarsPb'][key] = kwargs["SampledVarsPb"][key] if 'SampledVarsPb' in kwargs.keys() else 1.0
```

Comment:
dict comprehension would be much faster here

Reply:
changed.